orc/orcarm.c

```diff
@@ -268,12 +294,41 @@ orc_arm_emit_branch (OrcCompiler *compiler, int cond, int label)
 {
   orc_uint32 code;
 
-  code = 0x0afffffe;
-  code |= (cond&0xf) << 28;
+  if (compiler->is_64bit) {
+    /** B.cond
+     *    3                   2                   1
+     *  1 0 9 8 7 6 5 4 3 2 1 0 9 8 7 6 5 4 3 2 1 0 9 8 7 6 5 4 3 2 1 0
+     * +---------------------------------------------------------------+
+     * |0 1 0 1 0 1 0|0|                imm19                |0| cond  |
+     * +---------------------------------------------------------------+
+     *
+     *  B
+     *    3                   2                   1
+     *  1 0 9 8 7 6 5 4 3 2 1 0 9 8 7 6 5 4 3 2 1 0 9 8 7 6 5 4 3 2 1 0
+     * +---------------------------------------------------------------+
+     * |0 0 0 1 0 1|                       imm26                       |
+     * +---------------------------------------------------------------+
+     *
+     * Note that we don't know exact imm19/imm26 yet, so need fixup codes.
+     */
+    if (cond < ORC_ARM_COND_AL) {
+      code = 0x54000000;
+      code |=  cond&0xf;
+
+      ORC_ASM_CODE(compiler,"  b.%s .L%d\n", orc_arm_cond_name(cond), label);
+    } else {
+      code = 0x14000000;
+
+      ORC_ASM_CODE(compiler,"  b .L%d\n", label);
+    }
+  } else {
+    code = 0x0afffffe;
+    code |= (cond&0xf) << 28;
+
+    ORC_ASM_CODE(compiler,"  b%s .L%d\n", orc_arm_cond_name(cond), label);
```

Comment:
Is it ok to do ORC_ASM_CODE before orc_arm_add_fixup & orc_arm_emit ?

Reply:
Yes, `orc_arm_add_fixup` & `orc_arm_emit` access different variables in `compiler`.
The order is not important.

Reply:
Ok..